x-pack/plugins/apm/public/hooks/useTransactionCharts.ts

```diff
@@ -30,6 +31,7 @@ export function useTransactionCharts() {
               transactionType,
               transactionName,
               uiFilters: JSON.stringify(uiFilters),
+              environment: uiFilters.environment || ENVIRONMENT_ALL.value,
```

Comment:
Isn't `environment` now sent twice? Both as `query.uiFilters.environment` and as `query.environment`? 
This duplication could lead to inconsistencies if the values are not identical and we sometimes read from `query.uiFilters.environment` and other times from `query.environment`

Reply:
`environment` is duplicated since it's now a required query param, and also included in `uiFilters['service.environment']`, which is an optional. I could remove the default value `ENVIRONMENT_ALL.value` here to ensure it's always consistent. Otherwise, to remove the duplication, we would need to have `uiFilters` be required with `['service.environment'] also being required as a nested property. This would require validation of the uiFilters object after it's deserialized.

Reply:
> Otherwise, to remove the duplication, we would need to have uiFilters be required with `['service.environment'] also being required as a nested property. This would require validation of the uiFilters object after it's deserialized.

I think we should remove the duplication, either by removing environment from `uiFilters` (might be a bigger change) or remove the `environment` property you've added.
We don't strictly have to use io-ts for the validation if it makes it harder. We could also simply deserialize `uiFilters` and then `throw` if `uiFilters.environment` is not given.

Reply:
@dgieselaar might have some tricks up his sleeve for validating (serialised) json with io-ts.

Reply:
in the latest commit i removed the duplicate `environment` param and explicitly check the de-serialized `uiFilters` for `environment` and throw a `Boom.badRequest` if it's missing.